.groovylintrc.json

```diff
@@ -1,5 +1,5 @@
 {
-    "extends": "recommended",
+    "extends": "recommended-jenkinsfile",
```

Comment:
What is the difference between recommended and recommended-jenkinsfile?

Reply:
It's the linter's recommended (format)[https://github.com/nvuillam/npm-groovy-lint/tree/main#format] which lints jenkinsfiles too. When using just 'recommended' the linter catches errors that are not actually errors and breaks the program.

Reply:
Can you provide some examples what failures are you encountering? I am doubtful what to use as well. On high level looks like `recommended` is best path for groovy. `recommended-jenkinsfile` seems to be applicable for jenkinsFiles specifically? Please correct me if I am wrong. 

Adding @prudhvigodithi @peterzhuamazon @dblock to get some input.

Reply:
One error I saw was that 'variable x was not used in the file'. However, x was referenced inside a string. 

Running the linter's automatic fixing caused tests to fail if I used the 'jenkins' setting while it was able to build and test fine with 'jenkins-file'